docs/mine/mitigating-dos.md

```diff
@@ -0,0 +1,109 @@
+
+First, [set a static port for your Miner using the guide](/mine/setting-a-static-port)
+
+Then use the following command to block a specific port:
+
+```
+# Using port 80 to exemplify. This will block the port in all the network interfaces
+iptables -A INPUT -p tcp --destination-port 80 -j DROP
+```
+
+Alternatively, you can just set the default policy to drop packets in all ports and only accept in a few
+
+```
+# Set the INPUT policy to DROP:
+iptables -P INPUT DROP
+
+# Allow packets from connections related to established ones, packets from established ones, and packets from localhost:
+iptables -A INPUT -m state --state RELATED,ESTABLISHED -j ACCEPT
+iptables -A INPUT -i lo -j ACCEPT
+
+# Allow new connections to TCP ports 22 (for SSH) and the ports of your miner (here, using 12345 as example):
+iptables -A INPUT -p TCP -m multiport --dports 22,12345 -m state --state NEW -j ACCEPT
+```
+
+At the end of setting these rules, save them with:
+
+```
+# for Ubuntu
+sudo /sbin/iptables-save
+
+# for Red Hat / CentOS –
+sudo /sbin/service iptables save
+# or
+sudo  /etc/init.d/iptables save 
+```
+
+## IP based protection
+
+If you detect incoming traffic from an IP address, you can block all incoming traffic coming from that location. 
+
+```
+# Blocking 192.168.1.100 to exemplify.
+iptables -I INPUT -s 192.168.1.100 -j DROP
+```
+
+After setting this rule, remember to save your new rule with
+
+```
+# for Ubuntu
+sudo /sbin/iptables-save
+
+# for Red Hat / CentOS –
+sudo /sbin/service iptables save
+# or
+sudo  /etc/init.d/iptables save 
+```
+
+Learn more how to maintain a denylist of IPs at [Blocking IP Addresses in Linux with IPTables](https://linux-audit.com/blocking-ip-addresses-in-linux-with-iptables)
+
+## libp2p based protection
+
+If a malicious actor has successfully dialed to your miner and is spamming it at the Gossipsub level, you can block it at the libp2p level.
+
+To detect this, look for patterns such as:
+
+```
+2020-08-31T16:12:16.214-0400	INFO	pubsub	go-libp2p-pubsub@v0.3.5/pubsub.go:927	received message from router graylisted peer 12D3KooWFf6t5Zj4u8585r9bLwU3unykX4DmHftUtx6kEzw89anL. Dropping RPC
+2020-08-31T16:12:16.214-0400	INFO	pubsub	go-libp2p-pubsub@v0.3.5/pubsub.go:927	received message from router graylisted peer 12D3KooWJp8Kj9w65eQcsEdfT4dfhsWAeiroMfxAdZCzqd2wbHq2. Dropping RPC
+2020-08-31T16:12:16.215-0400	INFO	pubsub	go-libp2p-pubsub@v0.3.5/pubsub.go:927	received message from router graylisted peer 12D3KooWJp8Kj9w65eQcsEdfT4dfhsWAeiroMfxAdZCzqd2wbHq2. Dropping RPC
+```
+
+In this case, it is likely that a node is misbehaving and sending so many invalid messages that it was already discarded from the Gossipsub routing table. To disable it from successfully setting a libp2p connection, you can use the following command:
+
+```
+TODO: ADD COMMAND HERE
+```
```

Comment:
@vyzo what's the command to block a PeerId from Lotus (if any)

Reply:
I don't think we have any!

Reply:
We chatted over Slack, @vyzo will expose the Connection Manager gater through a Lotus command. ETA end of the week